src/entities/Pool.ts

```diff
@@ -138,6 +157,61 @@ export class Pool extends Entity {
     return this._query(null, () => this.network(chainId).pipe(switchMap((network) => network.vault(scId, asset))))
   }
 
+  /**
+   * Get the currency of the pool.
+   */
+  currency() {
+    return this._query(['currency'], () => {
+      return this._root._protocolAddresses(this.chainId).pipe(
+        switchMap(({ hubRegistry }) => {
+          return this._root.getClient(this.chainId)!.readContract({
```

Comment:
This just gives you an AssetId. If you do `new AssetId(result)`, you can check `aasetId`.isNationalCurrency. We should hardcode the name and symbol for those. If it's not a national currency we should fetch the name and symbol from the network where the asset is from. `assetId.centrifugeId` gives you the centrifuge ID. not sure yet how to get to the chain from there.